File: plugin/shared/src/main/java/com/jonathanpotts/cartographytable/shared/models/VectorXZ.java
package com.jonathanpotts.cartographytable.shared.models;

/**
 * Stores a 2D vector containing an X and Z coordinate.
 */
public class VectorXZ {
    /**
     * X coordinate.
     */
    public int x;

    /**
     * Z coordinate.
     */
    public int z;

    /**
     * Creates a 2D vector containing an X and Z coordinate.
     *
     * @param x X coordinate.
     * @param z Z coordinate.
     */
    public VectorXZ(int x, int z) {
        this.x = x;
        this.z = z;
    }
}

File: shared/src/commonMain/kotlin/io/github/jonathanpotts/cartographytable/shared/models/VectorXZ.kt
package io.github.jonathanpotts.cartographytable.shared.models

import kotlinx.serialization.Serializable

/**
 * Stores a 2D vector containing an X and Z coordinate.
 *
 * @property x X coordinate.
 * @property z Z coordinate.
 */
@Serializable
data class VectorXZ(
    var x: Int = 0,
    var z: Int = 0
)
